src/dodal/common/beamlines/beamline_utils.py

```diff
@@ -162,3 +162,15 @@ def set_path_provider(provider: PathProvider):
 
 def get_path_provider() -> PathProvider:
     return PATH_PROVIDER
```

Comment:
Should: Going to invoke [PEP20](https://peps.python.org/pep-0020/) here and say there should only be one way to get the path provider, so I think you should delete this and move everything that calls it over to your new function